packages/manager/modules/pci/src/projects/project/notebooks/add/add.html

```diff
@@ -1,13 +1,136 @@
+        Cras purus dolor, fermentum sit amet vulputate id, pretium quis lorem.
+    </p>
+</div>
+
+<!--Banner messages-->
 <div class="mt-4" id="addMessages">
+    <oui-message data-type="info">
+        <span data-translate="pci_notebook_add_banner_object_storage_description"></span>
+        <a
+            data-ui-sref="pci.projects.project.storages"
+            target="_blank"
+        >
+            <span
+                data-translate="pci_notebook_add_banner_object_storage_description_link"
+            ></span>
+        </a>
+    </oui-message>
+
     <cui-message-container
         data-messages="$ctrl.messages"
     ></cui-message-container>
 </div>
+
+<!--Notebook stepper-->
+<oui-stepper data-on-finish="$ctrl.onNotebookSubmit()">
+
+    <!--Notebook Configuration-->
+    <oui-step-form
+        name="notebook_settings"
+        data-header="{{:: 'pci_notebook_add_configuration_title' | translate }}"
+        data-on-focus="$ctrl.displayNotebookConfiguration = true"
+        data-on-submit="$ctrl.displayNotebookConfiguration = false"
+        data-prevent-next="true"
+    >
+        <notebook-configuration
+            data-display="$ctrl.displayNotebookConfiguration"
+            data-notebook="$ctrl.notebook"
+            data-editors="$ctrl.editors"></notebook-configuration>
+    </oui-step-form>
+
+    <!--Notebook Framework-->
+    <oui-step-form
+        data-name="notebook_framework"
+        data-header="{{:: 'pci_notebook_add_framework_title' | translate }}"
+        data-on-focus="$ctrl.displayNotebookFramework = true"
+        data-on-submit="$ctrl.displayNotebookFramework = false"
+        data-navigation="$ctrl.notebook.selected.framework.model"
+    >
+        <notebook-frameworks-list
+            data-display="$ctrl.displayNotebookFramework"
+            data-notebook="$ctrl.notebook"
+            data-frameworks="$ctrl.frameworks"></notebook-frameworks-list>
+    </oui-step-form>
+
+    <!--Notebook Privacy Settings-->
+    <oui-step-form
+        name="notebook_private_settings"
+        data-header="{{:: 'pci_notebook_add_privacy_settings_title' | translate }}"
+        data-on-focus="$ctrl.displayNotebookPrivacySettings = true"
+        data-on-submit="$ctrl.displayNotebookPrivacySettings = false"
+        data-navigation="$ctrl.notebook.selected.privacy"
+    >
+        <notebook-privacy-settings
+            data-display="$ctrl.displayNotebookPrivacySettings"
+            data-notebook="$ctrl.notebook"></notebook-privacy-settings>
+    </oui-step-form>
+
+    <!--Notebook Datacenter Location-->
+    <oui-step-form
+        name="notebook_datacenter_location"
+        data-header="{{:: 'pci_notebook_add_privacy_datacenter_location_title' | translate }}"
+        data-on-focus="$ctrl.displayNotebookDatacenterLocation = true"
+        data-on-submit="$ctrl.onLocationStepSubmit()"
```

Comment:
I am not able to find the definition for `onLocationStepSubmit` method :)

Reply:
oudate